integr8ly/grafana-operator
# fix-307-implement ready/live probe customization

## Description

- adding customization to grafana readiness/liveness probes
- This change will overwrite hardcoded values for these probes, if no values are defined within the CR then the operator will stick with default values

## Relevant issues/tickets

#307
## Type of change




## Checklist


## Verification steps

Checkout the master branch:
- run `make cluster/prepare/local NAMESPACE=grafana`
- run the operator locally with operator-sdk `operator-sdk run local --watch-namespace="" --operator-flags="--zap-level=1"`
- wait for the operator to properly deploy all resources
- check the grafana-deployment Pod yaml and note the values in Liveness and Readiness probe fields 
- stop the operator 
Once the above have finished:
- checkout this branch locally
- edit `deploy/examples/Grafana.yaml` and append the following to the end of the spec field:

where N is an arbitrary number (up to preference)
```
  livenessProbeSpec:
    initialDelaySeconds: N
    timeoutSeconds: N
    periodSeconds: N
    sucessThreshold: N
    failureThreshold: N
  readinessProbeSpec:
    initialDelaySeconds: N
    timeoutSeconds: N
    periodSeconds: N
    successThreshold: N
    failureThreshold: N
```
- apply the new Grafana.yaml cr
- ensure that the Liveness and Readiness probes are updated accordingly on the cluster
